src/components/ButtonIcon/index.d.ts

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react';
+import { BaseProps } from './../types';
+
+export interface ButtonIconProps extends BaseProps {
+    assistiveText?: string;
+    ariaControls?: string;
+    ariaExpanded?: boolean;
+    ariaHaspopup?: boolean;
+    ariaPressed?: boolean;
+    disabled?: boolean;
+    form?: string;
+    icon: JSX.ElementChildrenAttribute;
+    id?: string;
+    // onBlur?: ;
+    // onClick?: ;
+    // onFocus?: ;
+    // onKeyDown?: ;
+    shaded?: boolean;
+    size?: 'xx-small' | 'x-small' | 'small' | 'medium' | 'large';
```

Comment:
Would there be any benefit of type-aliasing these union types to share across files? I noticed there is a lot of repeat. I can see maintainability issues if we need to rename or add a new size in the future...

Type aliases are resolved before showing up in IDEs like VS Code, so they won't affect clients' workflow.